tfhe/src/core_crypto/experimental/algorithms/shared_lwe_secret_key_generation.rs

```diff
@@ -0,0 +1,67 @@
+//! Module containing primitives pertaining to shared [`LweSecretKey`] generation.
+
+use crate::core_crypto::commons::generators::SecretRandomGenerator;
+use crate::core_crypto::commons::math::random::{
+    ByteRandomGenerator, RandomGenerable, UniformBinary,
+};
+use crate::core_crypto::commons::numeric::Numeric;
+use crate::core_crypto::commons::parameters::LweDimension;
+use crate::core_crypto::commons::traits::ContainerMut;
+use crate::core_crypto::entities::{LweSecretKey, LweSecretKeyOwned};
+use crate::core_crypto::experimental::commons::parameters::LweSecretKeySharedCoefCount;
+use crate::core_crypto::experimental::commons::traits::SecretKey;
+
+/// Fill an [`LWE secret key`](`LweSecretKey`) with a predefined number of coefficients from an
+/// already existing secret key which can be smaller than the input key element count and fill the
+/// rest with random coefficients.
+pub fn generate_shared_binary_lwe_secret_key<Scalar, KeyCont, InputSecretKey, Gen>(
+    lwe_secret_key_with_shared_coefficients: &mut LweSecretKey<KeyCont>,
+    source_key_to_share_coefficients: &InputSecretKey,
+    lwe_secret_key_shared_coef_count: LweSecretKeySharedCoefCount,
+    generator: &mut SecretRandomGenerator<Gen>,
+) where
+    Scalar: RandomGenerable<UniformBinary> + Numeric,
+    KeyCont: ContainerMut<Element = Scalar>,
+    InputSecretKey: SecretKey<Scalar>,
+    Gen: ByteRandomGenerator,
+{
+    assert!(
```

Comment:
Same assert with `source_key_to_share_coefficients`

Reply:
not necessary with the new way of doing the shared lwe sk